agent/src/main/java/com/teamscale/jacoco/agent/Agent.java

```diff
@@ -71,42 +69,64 @@ public Agent(AgentOptions options,
 	protected void initServerEndpoints(Service spark) {
 		spark.get("/partition", (request, response) ->
 				Optional.ofNullable(options.getTeamscaleServerOptions().partition).orElse(""));
+		spark.get("/message", (request, response) ->
+				Optional.ofNullable(options.getTeamscaleServerOptions().message).orElse(""));
 		spark.post("/dump", this::handleDump);
 		spark.post("/reset", this::handleReset);
-		spark.post("/partition/" + PARTITION_PARAMETER, this::handleSetPartition);
+		spark.put("/partition", this::handleSetPartition);
+		spark.put("/message", this::handleSetMessage);
 	}
 
 	/** Handles dumping a XML coverage report for coverage collected until now. */
 	private String handleDump(Request request, Response response) {
 		logger.debug("Dumping report triggered via HTTP request");
 		dumpReport();
-		response.status(204);
+		response.status(HttpServletResponse.SC_NO_CONTENT);
 		return "";
 	}
 
 	/** Handles resetting of coverage. */
 	private String handleReset(Request request, Response response) {
 		logger.debug("Resetting coverage triggered via HTTP request");
 		controller.reset();
-		response.status(204);
+		response.status(HttpServletResponse.SC_NO_CONTENT);
 		return "";
 	}
 
 	/** Handles setting the partition name. */
 	private String handleSetPartition(Request request, Response response) {
-		String partition = request.params(PARTITION_PARAMETER);
+		String partition = request.body();
 		if (partition == null || partition.isEmpty()) {
-			logger.error("Partition missing in " + request.url() + "! Expected /partition/Some%20Partition%20Name.");
+			logger.error("The new partition name is missing in the request body! Please add it as plain text " +
+					"(content type \"text/plain; charset=UTF-8\"");
 
-			response.status(400);
+			response.status(HttpServletResponse.SC_BAD_REQUEST);
 			return "Partition name is missing!";
 		}
 
 		logger.debug("Changing partition name to " + partition);
 		controller.setSessionId(partition);
 		options.getTeamscaleServerOptions().partition = partition;
 
-		response.status(204);
+		response.status(HttpServletResponse.SC_NO_CONTENT);
+		return "";
+	}
+
+	/** Handles setting the partition name. */
+	private String handleSetMessage(Request request, Response response) {
+		String message = request.body();
+		if (message == null || message.isEmpty()) {
+			logger.error("The new message is missing in the request body! Please add it as plain text " +
+					"(content type \"text/plain; charset=UTF-8\"");
+
+			response.status(HttpServletResponse.SC_BAD_REQUEST);
+			return "Message is missing!";
```

Comment:
Would the error string not also make a got return value? Why be more verbose/helpful on only one channel?